apps/omg_child_chain/test/omg_child_chain/block_queue/core_test.exs

```diff
@@ -340,164 +348,218 @@ defmodule OMG.ChildChain.BlockQueue.CoreTest do
+               |> Core.set_ethereum_status(1, 1000, false)
+               |> elem(1)
+               |> Core.get_blocks_to_submit()
+    end
 
-      {_, state} =
-        state |> Core.enqueue_block(<<0>>, next_blknum, 1) |> Core.set_ethereum_status(1000, current_blknum, false)
+    test "All blocks enqueued get the same gas price", %{empty_high_max_gas_price: empty} do
+      assert [%{gas_price: 36_000_000_000}, %{gas_price: 36_000_000_000}] =
+               empty
+               |> Core.set_ethereum_status(0, 0, false)
+               |> elem(1)
+               |> Core.enqueue_block("1", 1000, 0)
+               |> Core.set_ethereum_status(1, 0, false)
+               |> elem(1)
+               |> Core.enqueue_block("2", 2000, 0)
+               |> Core.set_ethereum_status(2, 0, false)
+               |> elem(1)
+               |> Core.get_blocks_to_submit()
+    end
 
-      assert expected_price > state.gas_price_to_use
+    test "Gas price calculation cannot be raised above limit", %{empty_high_max_gas_price: empty} do
+      assert [%{gas_price: 40_000_000_000}] =
+               empty
+               |> Core.set_ethereum_status(0, 0, false)
+               |> elem(1)
+               |> Core.enqueue_block("1", 1000, 0)
+               |> Core.set_ethereum_status(1, 0, false)
+               |> elem(1)
+               |> Core.set_ethereum_status(2, 0, false)
+               |> elem(1)
+               |> Core.set_ethereum_status(4, 0, false)
+               |> elem(1)
+               |> Core.get_blocks_to_submit()
     end
 
-    test "gas price change only, when try to push blocks", %{empty_with_gas_params: state} do
-      gas_price = state.gas_price_to_use
+    test "Gas price calculation cannot be lowered below limit", %{empty_high_max_gas_price: empty} do
+      state =
+        empty
+        |> Core.set_ethereum_status(0, 0, false)
+        |> elem(1)
+        |> Core.enqueue_block("1", 1000, 0)
 
+      # many successful submissions
       state =
-        Enum.reduce(4..10, state, fn eth_height, state ->
-          {_, state} = Core.set_ethereum_status(state, eth_height, 0, false)
-          assert gas_price == state.gas_price_to_use
+        Enum.reduce(2..1000, state, fn eth_height, state ->
           state
+          |> Core.set_ethereum_status(eth_height - 1, (eth_height - 2) * 1000, false)
+          |> elem(1)
+          |> Core.enqueue_block("#{eth_height}", eth_height * 1000, eth_height - 2)
         end)
 
-      state = state |> Core.enqueue_block(<<0>>, 6000, 1)
-      {_, state} = Core.set_ethereum_status(state, 101, 0, false)
-      assert state.gas_price_to_use > gas_price
-    end
-
-    test "gas price changes only, when etherum advanses", %{empty_with_gas_params: state} do
-      gas_price = state.gas_price_to_use
-      eth_height = 0
+      assert [%{gas_price: 5}, %{gas_price: 5}] = Core.get_blocks_to_submit(state)
 
+      # no more successful submissions can change the price
```

Comment:
I don't follow the test logic atm. If I'm correct you are doing the exact same thing 2 more times (after prev `1000-2` trials) with a comment that now it can change the price. Final asserts shows the price hasn't changed :question:

Reply:
yeah, so the idea is to "reach" the bottom price (5) and do one more (actually I don't know why I did 2, a leftover, I'll fix this) and see it remain at 5